tasks.py

```diff
@@ -295,3 +295,38 @@ def write_file(full_path: Path, content: str):
                 full_path=docs_sub_dir.joinpath(md_filename),
                 content=template.format(name=basename, py_path=py_path),
             )
+
+
+@task(help={"filename": "The JSON file containing the OpenAPI schema to validate"})
+def run_swagger_validator(_, fname):
+    """This task can be used in the CI to test the generated OpenAPI schemas
+    with the online swagger validator.
+
+    Returns:
+        non-zero exit code if validation fails, otherwise returns 0.
+
+    """
+
+    import requests
+
+    swagger_url = "https://validator.swagger.io/validator/debug"
+    with open(fname, "r") as f:
+        schema = json.load(f)
+    response = requests.post(swagger_url, json=schema)
+
+    if response.status_code != 200:
+        raise RuntimeError(f"Server returned status code {response.status_code}.")
+
+    try:
+        json_response = response.json()
+    except json.JSONDecodeError:
+        raise json.JSONDecodeError(
+            f"Unable to parse validator response as JSON: {response}"
+        )(response.json().get("messages", "Failed to get error messages from response"))
+
+    if len(json_response) > 0:
+        raise RuntimeError(
+            response.json().get(
+                "messages", "Failed to get error messages from response"
+            )
```

Comment:
I think we should simply return the full JSON response here, also using `json_response` and not using the `json` method again. It should only be used once.